gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/sampling/PageRankSamplingTest.java

```diff
@@ -66,89 +63,67 @@ public SamplingAlgorithm getSamplingOperator() {
-      } else {
-        if (keepVerticesIfSameScore) {
-          assertEquals("not all vertices got sampled (should be, all got same score)",
-            dbVertices.size(), newVertices.size());
-        } else {
-          assertTrue("some vertices got sampled (should NOT be, all got same score)",
-            newVertices.isEmpty());
-        }
-      }
-      dbEdges.removeAll(newEdges);
-      for (Edge edge : dbEdges) {
-        assertFalse("edge from original graph was not sampled but source and target were",
-          newVertexIDs.contains(edge.getSourceId()) &&
-            newVertexIDs.contains(edge.getTargetId()));
       }
-
-      // test special graph
-      LogicalGraph specialGraph = getLoaderFromString(
-        "(alice:Person {name : \"Alice\"})\n" +
-          "(eve:Person {name : \"Eve\"})\n" +
-          "g0:Community {interest : \"Friends\", vertexCount : 2} [\n" +
-          "(eve)-[eka:knows {since : 2013}]->(alice)\n" +
-          "(alice)-[ake:knows {since : 2013}]->(eve)]").getLogicalGraphByVariable("g0");
-
-      LogicalGraph sampledGraph = getSamplingOperator().sample(specialGraph);
-
-      List<Vertex> specialVertices = Lists.newArrayList();
-      List<Vertex> specialSampledVertices = Lists.newArrayList();
-
-      specialGraph.getVertices().output(new LocalCollectionOutputFormat<>(specialVertices));
-      sampledGraph.getVertices().output(new LocalCollectionOutputFormat<>(specialSampledVertices));
-
-      getExecutionEnvironment().execute();
-
-      assertNotNull("graph was null", sampledGraph);
-
-      GraphHead specialGh = sampledGraph.getGraphHead().collect().get(0);
-      double minScore1 = specialGh.getPropertyValue(
-        SamplingConstants.MIN_PAGE_RANK_SCORE_PROPERTY_KEY).getDouble();
-      double maxScore1 = specialGh.getPropertyValue(
-        SamplingConstants.MAX_PAGE_RANK_SCORE_PROPERTY_KEY).getDouble();
-
-      assertEquals("min PageRankScore is not equal to max PageRankScore",
-        minScore1, maxScore1, 0.0);
-
+    } else {
       if (keepVerticesIfSameScore) {
-        assertEquals(
-          "special: not all vertices got sampled (should be, all got same score)",
-          specialVertices.size(), specialSampledVertices.size());
+        assertEquals("not all vertices got sampled (should be, all got same score)",
+          dbVertices.size(), newVertices.size());
       } else {
-        assertTrue(
-          "special: some vertices got sampled (should NOT be, all got same score)",
-          specialSampledVertices.isEmpty());
+        assertTrue("some vertices got sampled (should NOT be, all got same score)",
+          newVertices.isEmpty());
       }
-    } catch (Exception e) {
-      e.printStackTrace();
     }
+    dbEdges.removeAll(newEdges);
+    for (Edge edge : dbEdges) {
+      assertFalse("edge from original graph was not sampled but source and target were",
+        newVertexIDs.contains(edge.getSourceId()) &&
+          newVertexIDs.contains(edge.getTargetId()));
```

Comment:
Split in two assertions.

Reply:
How? It would be `assertFalse(A && B)` = `assertTrue(!(A && B))` = `assertTrue(!A || !B)`

Reply:
Oh I see ... that means all edges between sampled vertices have to be sampled too, right? Then this check is OK. Ignore my comment ..